java/piranha/src/test/java/com/uber/piranha/XPFlagCleanerTest.java

```diff
@@ -396,13 +396,120 @@ public void positiveCaseWithFlagNameAsVariable() throws IOException {
     }
   }
 
+  @Test
+  public void caseWithFlagNameAsVariableWithReturnType() throws IOException {
+    /* todo - piranha deletes the line
```

Comment:
given the flexibiity with the APIs, Piranha will need one more pass to delete the variable declarations [here](https://github.com/uber/piranha/blob/master/java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java#L710). While this is a good test to check for return type based deletion, this may be not happen in practice.

Reply:
Hmm, I'm not sure I followed. In this case, Piranha is deleting a variable declaration that is still in use. Is there an easy way to fix this?

Reply:
In the original implementation, all flag API invocations independent of their receiver/return types would be deleted and therefore deleting the flag deletion was not a problem. This is likely to be the case in real codebases also. 

However, for the test case here, given the differentiating functionality is being tested, the variable decl (which doesn't look at the uses) is incorrectly being deleted. I am okay with this change. Can you file an issue for this TODO? It has to be fixed by doing a two pass analysis. In the first pass, delete all the uses and if there are no uses for the variable, delete the variable in the second pass.

Reply:
:+1: